worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -47,6 +47,26 @@ function onBeforeProcessDocument(e) {
     e.cancel = ! eval(action.conditionFunction)(e.document, arguments);
 }
 
+function onProcessDocument(e) {
+    if (!e.application.getInputMessageProcessor().getProcessSubdocumentsSeparately()) {
+        traverseDocumentForSettingWorkerVersion(e.document);
+    } else {
+        setWorkerVersion(e.document);
+    }
+}
+
+function traverseDocumentForSettingWorkerVersion(document) {
+    setWorkerVersion(document);
+    for each(var subdoc in document.getSubdocuments()) {
+        setWorkerVersion(subdoc);
```

Comment:
Actually shouldn't this be `traverseDocumentForSettingWorkerVersion(subdoc)`?  As it is I think it only does one level of subdocuments.

Reply:
Yes :) thats silly mistake